sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -84,7 +84,7 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
           bucketId: Option[Int],
           dataSchema: StructType,
           context: TaskAttemptContext): OutputWriter = {
-        new JsonOutputWriter(path, bucketId, dataSchema, context)
+        new JsonOutputWriter(path, bucketId, new JacksonGenerator(dataSchema)(_), context)
```

Comment:
I thought it made more sense to apply the schema early since it's immutable over whole dataset while the writer is per partition. As you see in the JsonOutputWriter the writer is applied there. Not sure if this makes difference from perf point but it makes structure clearer (at least in my mind)